engine/overworld/map_objects.asm

```diff
@@ -2671,13 +2671,23 @@ _UpdateSprites::
 	jr z, .ok
 	ld b, 28 * SPRITEOAMSTRUCT_LENGTH
 .ok
-	ldh a, [hUsedSpriteIndex]
+	ldh a, [hUsedOAMIndex]
 	cp b
 	ret nc
+	ldh a, [hUsedWeatherSpriteIndex]
+	ld c, a
+	ldh a, [hUsedOAMIndex]
+	cpl
+	add (NUM_SPRITE_OAM_STRUCTS * SPRITEOAMSTRUCT_LENGTH) + 1
+	sub SPRITEOAMSTRUCT_LENGTH
```

Comment:
Don't `add` and then `sub` unless there's some tricky flag-setting reason for doing so. Also the `cpl` optimization is generally commented.
```suggestion
	; a = (NUM_SPRITE_OAM_STRUCTS - 1) * SPRITEOAMSTRUCT_LENGTH - a
	cpl
	add (NUM_SPRITE_OAM_STRUCTS - 1) * SPRITEOAMSTRUCT_LENGTH + 1
```